### ISSUE TITLE
[DOCS] example from configuring-tls-docker.asciidoc does not work and is out of date

### ISSUE THREAD

**Issue by hobo05** [elastic/elasticsearch#43695]
**Elasticsearch version:** Version: 7.2.0, Build: default/docker/508c38a/2019-06-20T15:54:18.811730Z, JVM: 12.0.1

**Docker Image:** docker.elastic.co/elasticsearch/elasticsearch:7.2.0

**Description of the problem including expected versus actual behavior**:

When trying to run `create-certs.yml` using docker-compose from `https://github.com/elastic/elasticsearch/blob/v7.2.0/docs/reference/security/securing-communications/configuring-tls-docker.asciidoc` I expect that the certificates will be created and saved to the current directory. What actually happens is as follows

```
$ docker-compose -f create-certs.yml up
ERROR: Invalid interpolation format for "user" option in service "create_certs": "$\{UID:-1000\}"
```

This can be fixed by setting `user` directly to `'1000'`.

The next error you'll get is

```
$ docker-compose -f create-certs.yml up
Creating create_certs ... done
Attaching to create_certs
create_certs    | WARNING: An illegal reflective access operation has occurred
create_certs    | WARNING: Illegal reflective access by org.bouncycastle.jcajce.provider.drbg.DRBG (file:/usr/share/elasticsearch/lib/tools/security-cli/bcprov-jdk15on-1.61.jar) to constructor sun.security.provider.Sun()
create_certs    | WARNING: Please consider reporting this to the maintainers of org.bouncycastle.jcajce.provider.drbg.DRBG
create_certs    | WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
create_certs    | WARNING: All illegal access operations will be denied in a future release
create_certs    | Exception in thread "main" java.nio.file.FileAlreadyExistsException: /usr/share/elasticsearch/config/certificates/certs/bundle.zip
create_certs    |       at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:94)
create_certs    |       at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
create_certs    |       at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
create_certs    |       at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
create_certs    |       at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:478)
create_certs    |       at java.base/java.nio.file.Files.newOutputStream(Files.java:222)
create_certs    |       at org.elasticsearch.xpack.security.cli.CertificateGenerateTool.fullyWriteFile(CertificateGenerateTool.java:456)
create_certs    |       at org.elasticsearch.xpack.security.cli.CertificateGenerateTool.generateAndWriteSignedCertificates(CertificateGenerateTool.java:405)
create_certs    |       at org.elasticsearch.xpack.security.cli.CertificateGenerateTool.execute(CertificateGenerateTool.java:185)
create_certs    |       at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86)
create_certs    |       at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:124)
create_certs    |       at org.elasticsearch.cli.Command.main(Command.java:90)
create_certs    |       at org.elasticsearch.xpack.security.cli.CertificateGenerateTool.main(CertificateGenerateTool.java:156)
create_certs    | bash: line 6: unzip: command not found
create_certs exited with code 0

```

This is because in this commit https://github.com/elastic/dockerfiles/commit/b0c83e71a9fe72e89d87e87cea36ce44f09eb6a5#diff-21c4e096c31ca33cf67e2ad16f6dac93 for 7.2.0, `unzip` was removed from being installed in the Dockerfile for the [Contexts for 7.2.0](https://github.com/elastic/dockerfiles/pull/15) PR

The workaround right now is setting the user to `root` and installing unzip in the command. 

On a separate note, this example is using `elasticsearch-certgen` which has been deprecated since 6.1 as documented https://github.com/elastic/elasticsearch/blob/ddafdb8ca1dd25fa16c6dbd9acd9763de7fc4738/docs/reference/commands/certgen.asciidoc. The team may want to consider switching out the command with `bin/elasticsearch-certutil cert --silent --pem --in config/certificates/instances.yml --keep-ca-key --ca-pass "" --out bundle.zip;` or something to that effect.

**Steps to reproduce**:

 1. Follow directions from https://www.elastic.co/guide/en/elasticsearch/reference/7.2/configuring-tls-docker.html to create the `instances.yml`.
2. Grab the contents of create-certs.yml from the page.  At the time of writing, this was what I copied.
    ```
    version: '2.2'

    services:
      create_certs:
        container_name: create_certs
        image: docker.elastic.co/elasticsearch/elasticsearch:7.2.0
        command: >
          bash -c '
            if [[ ! -d config/certificates/certs ]]; then
              mkdir config/certificates/certs;
            fi;
            if [[ ! -f /local/certs/bundle.zip ]]; then
              bin/elasticsearch-certgen --silent --in config/certificates/instances.yml --out config/certificates/certs/bundle.zip;
              unzip config/certificates/certs/bundle.zip -d config/certificates/certs;
            fi;
            chgrp -R 0 config/certificates/certs
          '
        user: $\{UID:-1000\}
        working_dir: /usr/share/elasticsearch
        volumes: ['.:/usr/share/elasticsearch/config/certificates']
    ```
 3. Run `docker-compose -f create-certs.yml up`
 4. You should see it fail with the first error mentioned above.

**Comment by henningandersen** [comment]
Might be related to #43696 
Ping @dliappis

**Comment by dliappis** [comment]
@hobo05 First of all thank you for the very thorough issue and analysis.

I've opened #43748 to address the issues you pointed out.

**Comment by dliappis** [comment]
Closed by #43748
